Added the recursive-memoized approach for coin change combination pro…

mandy8055 opened the pull request:
…blem

# Welcome to JavaScript community

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)



### **Describe your change:**

* [ v] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?


### **Checklist:**
* [v ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [ v] This pull request is all my own work -- I have not plagiarized.
* [v ] I know that pull requests will not be merged if they fail the automated tests.
* [v ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [v ] All new JavaScript files are placed inside an existing directory.
* [v ] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [ v] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [v ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

raklaptudirm commented on Dynamic-Programming/CoinChange.js:
```diff
@@ -38,8 +81,24 @@ function minimumCoins (coins, amount) {
 function main () {
```
This is not a must do change. If you are free, you can change this. I want you to remove the main function and instead implement a `jest` test file.

mandy8055 replied:
Yes, you're right. It is not a must do change. I wanted to add the memoized approach for the above standard problem of Dynamic programming, so that readers could get both the approaches for solving the same.

Okay, I'll remove the `main` function and add a **jest spec file** instead.
Thanks

mandy8055 commented:
Hi @raklaptudirm,
As you mentioned; I added 2 base tests and 5 main tests for CoinChange Problem. I Refactored the code a bit and removed the Memoized approach as it was not necessary or must do.
